src/lib/trajectory/parse.ts

```
@@ -241,37 +241,165 @@ const parse_torch_sim_hdf5 = async (
+            break
+          }
+        }
+
+        if (atomic_numbers) break
+      }
+
+      // If still no atomic numbers found, try to infer from filename or make a reasonable guess
+      if (!atomic_numbers && filename && filename.includes(`water`)) {
+        // For water cluster, assume it's water molecules (H2O)
+        // We need to check the positions to determine how many atoms there are
+        if (positions && positions[0]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)